app/views/handlers/list_action.py

```diff
@@ -47,8 +49,33 @@ def handle_post(self):
             self.parent_location.block_id
         )
         self.questionnaire_store_updater.remove_answers(answer_ids_to_remove)
+        self.evaluate_sections()
         self.questionnaire_store_updater.save()
 
+    def evaluate_sections(self):
+        sections_to_evaluate = self._sections_to_evaluate()
+        section_paths = self._routing_paths(sections_to_evaluate)
+        for section_path in section_paths:
+            self._update_section_completeness(
+                Location(section_path[0][0]), section_path[1]
+            )
+
+    def _sections_to_evaluate(self):
+        list_name = self._schema.get_block(self.rendered_block['parent_id'])['for_list']
+        associated_sections = self._schema.get_sections_associated_to_list_name(
+            list_name
+        )
+        associated_sections = {(element, None) for element in associated_sections}
```

Comment:
This would mean it won't work with repeats?

So I had a block within a repeat that is only displayed when there is at least 2 members in the household.
The block wasn't displayed when there was 1 householder. Once the section for that individual was complete, I added another person, which should make the individual section be marked as in progress however it was still marked as complete.
Clicking View answers then results in a circular redirect.

Reply:
Fixed in the next push